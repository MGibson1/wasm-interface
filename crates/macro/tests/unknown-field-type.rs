use wasm_interfacegen_macro::JsInterface;

pub fn main(){}

#[derive(JsInterface)]
struct VecField {
    vec_field: Vec<String>
}

#[derive(JsInterface)]
struct NonInterfacedField {
    non_interfaced_field: SomeOtherStruct
}

struct SomeOtherStruct{}
