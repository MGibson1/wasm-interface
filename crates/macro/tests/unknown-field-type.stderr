error[E0277]: the trait bound `Vec<String>: __JsInterface` is not satisfied
 --> tests/unknown-field-type.rs:7:16
  |
7 |     vec_field: Vec<String>
  |                ^^^^^^^^^^^ the trait `__JsInterface` is not implemented for `Vec<String>`
  |
note: required by a bound in `_::{closure#0}::assert_impl_all`
 --> tests/unknown-field-type.rs:5:10
  |
5 | #[derive(JsInterface)]
  |          ^^^^^^^^^^^ required by this bound in `_::{closure#0}::assert_impl_all`
  = note: this error originates in the macro `wasm_interfacegen::assert_impl_all` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `SomeOtherStruct: __JsInterface` is not satisfied
  --> tests/unknown-field-type.rs:12:27
   |
12 |     non_interfaced_field: SomeOtherStruct
   |                           ^^^^^^^^^^^^^^^ the trait `__JsInterface` is not implemented for `SomeOtherStruct`
   |
note: required by a bound in `_::{closure#0}::assert_impl_all`
  --> tests/unknown-field-type.rs:10:10
   |
10 | #[derive(JsInterface)]
   |          ^^^^^^^^^^^ required by this bound in `_::{closure#0}::assert_impl_all`
   = note: this error originates in the macro `wasm_interfacegen::assert_impl_all` (in Nightly builds, run with -Z macro-backtrace for more info)
